go-controller/pkg/node/iprulemanager/ip_rule_manager.go

```diff
@@ -85,6 +85,10 @@ func (rm *Controller) Delete(rule netlink.Rule) error {
 	}
 	if reconcileNeeded {
 		return rm.reconcile()
+	} else {
+		// though we don't need to reconcile because we don't manage this rule, we should still ensure its deleted
+		// from the system. No need for full reconcile.
+		_ = netlink.RuleDel(&rule)
```

Comment:
The commit message talks about deleting ip rules that the controller doesn't manage. Wouldn't this be a problem  if the rule is some custom rule created by the admin? Or does the controller only delete rules that have specific attributes so we know RuleManager was the original controller of this rule?

Reply:
The rule manager doesnt have to own the rule to delete it here. Users of rule manager may use it exclusively to manager rules and it may find a stale rule and want to delete it.